dsp/primitives/demonstrate.py

```diff
@@ -74,85 +79,78 @@ def do_augment(train, k=None, return_all=False):
     return do_augment
 
 
-def sample(train, k: int):
-    branch_idx = (
-        hasattr(dsp.settings, 'branch_idx')
-        and isinstance(dsp.settings.branch_idx, int)
-        and dsp.settings.branch_idx
-    )
-    branch_idx = branch_idx or 0
-    # print(f"branch_idx = {branch_idx}")
-
-    rng = random.Random(branch_idx)
-    shuffled_train = list(train)
+def sample(train: list[Example], k: int):
+    """Sample k examples from train."""
+    rng = random.Random(dsp.settings.branch_idx)
+    shuffled_train = [example.copy() for example in train]
     rng.shuffle(shuffled_train)
 
-    subset = shuffled_train[:k]
-    subset = [dsp.Example(x) for x in subset]  # augmented=False
```

Comment:
reviewer note: shuffle_train does not affect the input type, so Example() conversion is unnecessary.